graspologic/plot/plot_matrix.py

```diff
@@ -0,0 +1,735 @@
+    return ax
+
+
+def sort_meta(length, meta, group_class, class_order=["size"], item_order=None):
+    """
+    Sort the data and metadata according to the sorting method
+
+    Parameters
+    ----------
+    length : int
+        Number of nodes
+    meta : pd.DataFrame, pd.Series, list of pd.Series or np.array, optional
+        Metadata of the matrix such as class, cell type, etc., by default None
+    group_class : list or np.ndarray, optional
+        Metadata to group the graph in the plot, by default None
+    class_order : str, optional
+        Attribute of the sorting class to sort classes within the graph, by default "size"
+    item_order : string or list of string, optional
+        Attribute in meta by which to sort elements within a class, by default None
+
+    Returns
+    -------
+    inds :
+        The index of the data after sorting
+    meta :
+        The metadata after sorting
+    """
+
+    # if metadata does not exist, return the original data
+    if meta is None or len(meta) == 0:
+        return np.arange(length), meta
+
+    meta = meta.copy()
+    # total_sort_by keeps track of what attribute was used to sort
+    total_sort_by = []
+    # create new columns in the dataframe that correspond to the sorting order
+    # one problem with this current sorting algorithm is that we cannot sort
+    # classes by size and other class attributes at the same time.
+
+    for gc in group_class:
+        for co in class_order:
+            if co == "size":
+                class_size = meta.groupby(gc).size()
+                # map each node from the sorting class to their sorting class size
+                meta[f"{gc}_size_order"] = meta[gc].map(class_size)
+                total_sort_by.append(f"{gc}_size_order")
+            else:
+                class_value = meta.groupby(gc)[co].mean()
+                # map each node from the sorting class to certain sorting class attribute
+                meta[f"{gc}_{co}_order"] = meta[gc].map(class_value)
+                total_sort_by.append(f"{gc}_{co}_order")
+            total_sort_by.append(gc)
+    total_sort_by += item_order
+
+    # arbitrarily sort the data from 1 to length
+    meta["sort_idx"] = range(len(meta))
+    # if we actually need to sort, sort by class_order, sort_class, item_order
+    if len(total_sort_by) > 0:
+        meta.sort_values(total_sort_by, inplace=True, kind="mergesort")
+
+    inds = np.copy(meta["sort_idx"].values)
+    meta["sort_idx"] = range(len(meta))
+    return inds, meta
+
+
+def matrixplot(
+    data,
+    ax=None,
+    meta=None,
+    plot_type="heatmap",
+    group_class=None,
+    class_order="size",
+    item_order=None,
+    color_class=None,
+    highlight=None,
+    palette="tab10",
+    ticks=True,
+    tick_pad=None,
+    color_pad=None,
+    border=True,
+    center=0,
+    cmap="RdBu_r",
+    sizes=(5, 10),
+    square=True,
+    gridline_kws=None,
+    spinestyle_kws=None,
+    highlight_kws=None,
+    **kws,
```

Comment:
not sure why you took out the possibility for separate row/col information? can you explain?

Reply:
I can add it back if that's a necessary feature, but I just removed it because I thought it was unnecessary and would be inconvenient to use the function

Reply:
it is a necessary feature.

Reply:
that was the logic behind having `adjplot` as well if one doesn't need those features.

Reply:
okay cool, I will do that then.

Reply:
Since I am doing separate row/col information, should I have documentation for each of the row/col information, or can I have one documentation per pair in matrixplot?

Reply:
refer to seaborn clustermap for an example of what it should look like